### DISCUSSION TITLE
Удаление дупликатов

### DISCUSSION THREAD

**Discussion by dablyt** [Chimildic/YaMuTools#104] [Вопрос/ответ]
Привет! Я решил перейти с ВК на Яндекс Музыку и для этого мне понадобилось перенести все мои плейлисты, а их очень много. Я прибегнул к способу, в котором добавляю все треки из плейлиста в ВК в "Мою музыку", а затем использую актуальную страницу импорта музыки в Яндекс Музыку. Из-за того, что в ВК очень много "пиратских" треков, залитых пользователями, и у данных сервисов разные базы, из 150 треков перенеслись 60 :)
Я решил оставшиеся 90 загрузить как MP3, но проблема в том, что сидеть и вручную смотреть, какие треки перенеслись, а какие нет, в каждом плейлисте - нереально. Я подумал, что загружу-ка я в этот плейлист скачанные 150 из ВК и удалю дупликаты. Тогда я нашёл твоё чудо-расширение, которое мне очень нравится своими перспективами и нынешними возможностями (надеюсь, дальше будет круче!), нашёл эту великолепную функцию "Поиск дупликатов", но, блин, как я обломался, когда обнаружил, что эта функция не сможет удалить мои загруженные 60 дупликатов, а всего лишь перенесёт их наверх. Автор, умоляю, придумай что нибудь (например, настройку Поиска дупликатов "Удалять с одинаковым названием"), ведь, я уверен, я не один такой, кто также голову себе ломает, как ему перенести несколько тысяч треков с одной платформы на другую. С меня донатик с подписью <3

**Comment by Chimildic** [comment]
Мне, мягко сказать, надоело делать одно и то же для разных сервисов/платформ. Поэтому вряд ли обновлю расширение. Учитывая появление нового дизайна сайта, его и вовсе нужно переписывать. 

У Яндекса сильно урезанная библиотека из-за отзыва прав на треки. Их могут подмешивать, например, в волну, но из публичного поиска их не найти. ВК по-прежнему живет на пиратах. Поэтому перенос 60 из 150 далеко не факт, что из-за названий от пользователей. Мне кажется, дольше, но проще все выкачать и загрузить файлами. Сохранится и полнота, и дубликатов не появится. 

Также вы можете попробовать проект [gooex](https://chimildic.github.io/gooex/). У него есть функция удаления дубликатов [Filter.dedupTracks](https://chimildic.github.io/gooex/reference/filter#deduptracks). Но она сравнивает лишь по целому вхождению имени. Задача сравнения с опечатками или ином написании в gooex не реализована. Однако раз YaMuTools поднимает дубликаты наверх, gooex тоже должен их поймать.

**Comment by dablyt** [comment]
Я не совсем разобрался как начать работать с gooex даже по гайду по вашей ссылке. Я понял, что это аналог goofy для Яндекса, но это мало что дало) Всё равно спасибо! Попробую углубиться и, думаю, получится.
